src/components/EnvelopeList.vue

```diff
@@ -290,32 +293,41 @@ export default {
 						})
 					}
 				}
-				logger.info(`deleting message ${envelope.databaseId}`)
-				try {
-					await this.$store.dispatch('deleteMessage', {
-						id: envelope.databaseId,
-					})
-				} catch (error) {
-					showError(await matchError(error, {
-						[NoTrashMailboxConfiguredError.getName()]() {
-							return t('mail', 'No trash mailbox configured')
-						},
-						default(error) {
-							logger.error('could not delete message', error)
-							return t('mail', 'Could not delete message')
-						},
-					}))
-				}
 			})
 
 			// Get new messages
 			this.$store.dispatch('fetchNextEnvelopes', {
 				mailboxId: this.mailbox.databaseId,
 				query: this.searchQuery,
-				quantity: this.selectedEnvelopes.length
+				quantity: this.selectedEnvelopes.length,
 			})
 
+			logger.info('deleting messages')
+			this.deleting = true
 			this.unselectAll()
+
+			try {
+				// Delete messages per batch of 50 messages so as to not overload server or create timeouts
+				let start = 0
+				let batch = []
+				do {
+					batch = ids.slice(start, start + 50)
+					await this.$store.dispatch('deleteMessages', { ids: batch })
+					start += 50
+				} while (batch.length === 50)
```

Comment:
doesn't that mean if you delete a multiple of 50 messages then there is always one additional request that deletes a batch of 0 ids?

you could copy the ids into a new array like ``const idsToDelete = [...ids]``and then you take your slices with ``const batch = idsToDelete.splice(50)`` run the loop ``while(idsToDelete.length > 0)``.

```js
const idsToCopy = [...ids]
while (idsToCopy.length > 0) {
    const batch = idsToCopy.splice(50)
    await this.$store.dispatch('deleteMessages', { ids: batch })
}
```

Reply:
copying to a new array (const idsToCopy = [...ids]) doesn't seem necessary neither

Reply:
done